dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/system/BroadcastReceiverSystemInfoProvider.kt

```diff
@@ -23,15 +23,17 @@ internal class BroadcastReceiverSystemInfoProvider :
     // region BroadcastReceiver
 
     override fun onReceive(context: Context, intent: Intent?) {
-        val action = intent?.action
-        when (action) {
+        when (val action = intent?.action) {
```

Comment:
I didn't know such expression is possible in Kotlin

Reply:
Was a suggestion from `AS` :)